sdk/core/Azure.Core/src/Pipeline/HttpPipelineOptions.cs

```diff
@@ -10,30 +10,26 @@ namespace Azure.Core.Pipeline
 {
     public class HttpClientOptions
     {
-        private HttpPipelineTransport _transport = HttpClientTransport.Shared;
+        public static string TelemetryPolicy { get; } = "Telemetry";
```

Comment:
Should we consider these to just be `public const` members?   Does making it a read-only static property offer a benefit that I'm overlooking?

Reply:
No, it can be whatever. I don't like consts because of embedding behaviour and JIT can optimize readonly property access out.

Reply:
Interesting.  I’m not familiar with this optimization and would like to learn.  :)

Are you saying that it will optimize the property access out and inline the reference or that it can remove the member during JIT if it wasn’t called?  

Do you know the reason why same optimization isn’t available for constants?